src/Bundle/AutoMapperBundle/CacheWarmup/CacheWarmer.php

```diff
@@ -40,20 +35,24 @@ public function isOptional()
 
     public function warmUp($cacheDir)
     {
-        $mapperClasses = [];
-
         foreach ($this->cacheWarmerLoaders as $cacheWarmerLoader) {
             foreach ($cacheWarmerLoader->loadCacheWarmupData() as $cacheWarmupData) {
-                $mapperClasses[] = $this->fileLoader->saveMapper(
-                    $this->mapperConfigurationFactory->create(
-                        $this->autoMapperRegistry,
-                        $cacheWarmupData->getSource(),
-                        $cacheWarmupData->getTarget()
-                    )
-                );
+                $this->autoMapperRegistry->getMapper($cacheWarmupData->getSource(), $cacheWarmupData->getTarget());
             }
         }
 
-        return $mapperClasses;
+        // preloaded files must be in cache directory
+        if (0 !== strpos($this->autoMapperCacheDirectory, $cacheDir)) {
+            return [];
+        }
+
+        $mapppers = array_keys(require_once sprintf('%s/registry.php', $this->autoMapperCacheDirectory));
+
+        return array_map(
+            function ($mapper) {
+                return sprintf('%s/%s.php', $this->autoMapperCacheDirectory, $mapper);
+            },
+            $mapppers
+        );
```

Comment:
I think it's better to return php files than just class names, since I'm not sure preload mecanism will be able to find the class just with its name, since classes are in global namespace